pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -509,3 +528,55 @@ func (s *Service) getInstanceENIs(instanceID string) ([]*ec2.NetworkInterface, e
 
 	return output.NetworkInterfaces, nil
 }
+
+func (s *Service) getImageRootDevice(imageID string) (*string, error) {
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []*string{aws.String(imageID)},
+	}
+
+	output, err := s.scope.EC2.DescribeImages(input)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(output.Images) == 0 {
+		return nil, errors.Errorf("no images returned when looking up ID %q", imageID)
+	}
+
+	return output.Images[0].RootDeviceName, nil
+}
+
+func (s *Service) getInstanceRootDeviceSize(instance *ec2.Instance) (*int64, error) {
+
+	for _, bdm := range instance.BlockDeviceMappings {
+		if aws.StringValue(bdm.DeviceName) == aws.StringValue(instance.RootDeviceName) {
+			input := &ec2.DescribeVolumesInput{
+				VolumeIds: []*string{bdm.Ebs.VolumeId},
+			}
+
+			out, err := s.scope.EC2.DescribeVolumes(input)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(out.Volumes) == 0 {
+				return nil, errors.Errorf("no volumes found for id %q", aws.StringValue(bdm.Ebs.VolumeId))
+			}
+
+			return out.Volumes[0].Size, nil
+		}
+	}
+	return nil, nil
+}
+
+func (s *Service) sdkToInstance(v *ec2.Instance) (*v1alpha1.Instance, error) {
```

Comment:
I'm not a huge fan of this function... but I also did not want to make converters.SDKToInstance a method of service... open to any suggestions 😢

Reply:
Hmm, could we get the volume information when we do a Describe call instead of having a method for it? I think there should be only 2 places where this happens.

Reply:
As in the Describe for the instance? Unfortunately that call returns everything about the volumes EXCEPT size. :(

Reply:
Got it, the only concern I have is that now there are two functions to convert an SDK ec2 instance to our own instance type, which might be confusing. Maybe let's leave a comment somewhere?

Reply:
It is definitely confusing. I'll add some comments. The thing that bugs me the most is that the original `converters.SDKToInstance` feels dishonest now, because the "our" instance that is returned by that is now incomplete, and not really usable without also sugaring in the Volume information.

Reply:
Updated. Comment language is hard :sob:.

Reply:
One concern here is that now the SDKToInstance that will give a complete instance is private while the SDKToInstance that will give an incomplete instance is public.

Does converters.SDKToInstance have any use now? Should that method be private and this one be public? I think we should also add a comment to converters.SDKToInstance that outline why that is insufficient for a complete instance representation.

Reply:
I wonder if maybe converters.SDKToInstance's logic should just be inside service.SDKToInstance and that method should be made public. converters being private wouldn't do us any good I think?

Reply:
That makes sense since converting from sdk to instance now *requires* a call to the ec2 service

Reply:
Ok so I made the above change. I'm not sure if/how we'd deprecate converters.SDKToInstance? I left it in there with a comment about how it's insufficient.